client/components/accordion/docs/example.jsx

```diff
@@ -10,15 +10,19 @@ import React from 'react';
 /**
  * Internal dependencies
  */
-import Accordion from 'components/accordion';
-import Gridicon from 'components/gridicon';
+import AccordionComponent from 'components/accordion';
+import GridiconComponent from 'components/gridicon';
 
+// Wrapper with `displayName` for proper display in devdocs
+const Accordion = props => <AccordionComponent { ...props } />;
 Accordion.displayName = 'Accordion';
+const Gridicon = props => <GridiconComponent { ...props } />;
 Gridicon.displayName = 'Gridicon';
```

Comment:
Do we really need to give a `displayName` to the `Gridicon` or `Accordion` component itself? To make devdocs UI show correct names, I think only the `AccordionExample` component should have a `displayName`?

Reply:
That did seem to be required for this component to work correctly. It's necessary for `GridIcon` to render here:

![Screen Shot 2020-01-21 at 13 34 06](https://user-images.githubusercontent.com/841763/72805299-d662b780-3c52-11ea-9aed-17b487f08d53.png)

and I recall it throwing errors as I worked on these fixes.

Reply:
I think that's related to the way the playground works.

Reply:
This is strange. `react-live` needs to somehow map a component name you type into the editor (i.e., `<JetpackLogo />` or `<Button />`) onto the minified name. That should be done by the `playground-scope` module and the `<LiveProvider scope>` prop.

If that didn't work reliably, it would mean that all components that are supported in the playground (see the list of reexports in `playground-scope`) would need a `displayName` field.

Could you try to reproduce what exactly were the errors thrown when these components wrappers weren't present?

Reply:
Update: I think I see these errors (failing to render `Gridicon`) now in my local Calypso.

Reply:
I'd speculate the names come from here:

https://github.com/Automattic/wp-calypso/blob/c528da6820b6c81a7aecfc6dbf3a0dc780d01349/client/devdocs/design/playground-scope.js#L75

Perhaps we should be using those playground scoped components in our examples?

Reply:
The trouble with `Gridicon` is that it's a return value of `React.memo( React.forwardRef( ... ) )` and then the `react-element-to-jsx-string` library can't figure out the display name of such a thing.

The library's `format` function accepts a custom `displayName` function as an option. And we could solve the problem with a custom implementation that takes `playground-scope` into account. I'll try to do it.